examples/discovery-search-app/cypress/integration/collection_filter/spec.ts

```diff
@@ -0,0 +1,142 @@
+  beforeEach(() => {
+    //TODO: swap this section with the helper function, once that code is merged in
+    // Sets up and handles the collections, component settings, and initial query requests that run on page-load
+    cy.server();
+    cy.fixture('collections/collections.json').as('collectionsJSON');
+    cy.route('GET', '**/collections?version=2019-01-01', '@collectionsJSON').as('getCollections');
+    cy.fixture('component_settings/componentSettings.json').as('componentSettingsJSON');
+    cy.route('GET', '**/component_settings?version=2019-01-01', '@componentSettingsJSON').as(
+      'getComponentSettings'
+    );
+    cy.fixture('query/query.json').as('queryJSON');
+    cy.route('POST', '**/query?version=2019-01-01', '@queryJSON').as('postQuery');
+    cy.visit('/');
+    cy.wait(['@getCollections', '@getComponentSettings', '@postQuery']);
+
+    // Set up/override routes & fixtures that are specific to this file
+    cy.fixture('query/singleCollectionQuery.json').as('singleCollectionQueryJSON');
+    cy.fixture('query/doubleCollectionQuery.json').as('doubleCollectionQueryJSON');
+  });
+
+  describe('when the example app loads', () => {
+    it('the collection facet select does not appear', () => {
+      cy.get('.collection-facet-select').should('not.exist');
+    });
+  });
+
+  describe('when a query is made', () => {
+    beforeEach(() => {
+      cy.get('.bx--search-input').type('abil{enter}');
+      cy.wait('@postQuery');
+    });
+
+    it('the collection selector should appear', () => {
+      cy.get('#collection-facet-select').should('exist');
+      cy.get('#collection-facet-select')
+        .contains('Available collections')
+        .should('exist');
+    });
+
+    describe('and we click the collection filter', () => {
+      beforeEach(() => {
+        cy.get('#collection-facet-select').click();
+      });
+
+      it('displays a list of the collections in the project', () => {
+        cy.get('#collection-facet-select__menu')
+          .contains('deadspin')
+          .should('exist');
+        cy.get('#collection-facet-select__menu')
+          .contains('espn')
+          .should('exist');
+        cy.get('#collection-facet-select__menu')
+          .contains('finnegans wake')
+          .should('exist');
+      });
+
+      describe('and we select a single collection', () => {
+        beforeEach(() => {
+          cy.route('POST', '**/query?version=2019-01-01', '@singleCollectionQueryJSON').as(
+            'postQuerySingleCollection'
+          );
+          cy.get('.bx--list-box__menu-item')
+            .contains('finnegans wake')
+            .click();
+          cy.wait('@postQuerySingleCollection').as('singleCollectionQueryObject');
+        });
+
+        it('should make a query against only the selected collection', () => {
+          cy.get('@singleCollectionQueryObject')
+            //@ts-ignore
+            .its('requestBody.collection_ids')
+            .should('contain', 'paris19221939');
```

Comment:
Is there a way to check that it contains only this collection_id? Below, I noticed you chained something similar with `and`, so it seems like this could technically be querying against multiple collections but we're only checking that it includes this one and not also that it doesn't include any others.

Reply:
I could do either
```
.should('eq', ['paris19221939']);
```
or
```
.should('contain', 'paris19221939')
.and('have.length', 1);
```

The former is probably cleaner

Reply:
Yeah, the former looks clear to me 👍